### PR TITLE
Sync on-chain tools with protocol front-end

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR

- Adds support for updating already minted components
- Adds support for using ERC20 tokens as service bonds
- Adds tests for newly added functionalities
- Updates documentation for the latest features
- Adds a list of addresses across various chains in documentation

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

File: tests/test_autonomy/test_chain/test_addresses.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+import pytest
+import requests
+
+from autonomy.chain import constants
+from autonomy.chain.config import ChainType
+
+
+ADDRESS_FILE_URL = "https://raw.githubusercontent.com/valory-xyz/autonolas-registries/main/docs/configuration.json"
+CHAIN_SLUGS = {ChainType.ETHEREUM: "mainnet", ChainType.GOERLI: "goerli"}
```
Comment (kupermind) [inline tests/test_autonomy/test_chain/test_addresses.py:32]: Please make a note somehow to account for other chains once they are available in the app.

File: autonomy/chain/constants.py
Comment (mariapiamo) [inline autonomy/chain/constants.py:1]: You can correct the comments regarding the source file of the contract addresses. From now on, we are going to use`autonolas-registries/main/scripts/deployment//main/docs/configuration.json for all the chains

**Review by mariapiamo — APPROVED** [review]

File: tests/test_autonomy/test_chain/test_addresses.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        chain_configs = requests.get(url=ADDRESS_FILE_URL).json()
+        cls.contracts = {
+            config["name"]: config["contracts"] for config in chain_configs
+        }
+
+    @pytest.mark.parametrize(
+        argnames="chain",
+        argvalues=(
+            ChainType.ETHEREUM,
+            ChainType.GOERLI,
```
Comment (DavidMinarsch) [inline tests/test_autonomy/test_chain/test_addresses.py:64]: What about the other chains?

File: tests/test_autonomy/test_chain/test_addresses.py (hunk 1 above)
Comment (DavidMinarsch) [reply tests/test_autonomy/test_chain/test_addresses.py:32]: I am confused. Where are the tests for Gnosis and Polygon? The CLI should support them already no?

Comment (angrybayblade) [reply tests/test_autonomy/test_chain/test_addresses.py:32]: They're supported via the ENV vars

File: tests/test_autonomy/test_chain/test_addresses.py (hunk 2 above)
Comment (angrybayblade) [reply tests/test_autonomy/test_chain/test_addresses.py:64]: https://github.com/valory-xyz/open-autonomy/pull/2030#discussion_r1318304519

**Review by 0xArdi — APPROVED** [review]

**Pull request merged**